layout/FallbackQuery.js

```diff
@@ -410,13 +403,31 @@ Layout.prototype.render = function( vs ){
 Layout.base = function( vs ){
   return {
     query: {
-      bool: {
-        should: []
+      function_score: {
```

Comment:
since this base layout is becoming more complex, it would be great to see it broken out into a separate file, especially given how lengthy this file happens to be.